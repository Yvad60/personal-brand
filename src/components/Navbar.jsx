import React from 'react';
import logo from '../assets/logo.svg';

const navigationLinks = ['Home', 'Blog', 'About'];
function Navbar() {
	return (
		<div className="w-full bg-primary_light flex text-primary_dark font-bold text-xl justify-between py-5 shadow-main px-20">
			<img src={logo} alt="logo" />
			<ul className="w-1/2 flex justify-between">
				{navigationLinks.map((link) => (
					<li key={link}>{link}</li>
				))}
				<button type="button">Contact</button>
			</ul>
		</div>
	);
}

export default Navbar;
