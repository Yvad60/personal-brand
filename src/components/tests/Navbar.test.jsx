import React from 'react';
import { render, screen } from '@testing-library/react';
import Navbar from '../Navbar.jsx';

describe('Navbar test', () => {
	it('should render the Navbar without crashing', () => {
		const wrapper = render(<Navbar />);
		expect(wrapper).toMatchSnapshot();
	});
});

describe('content tests', () => {
	beforeEach(() => {
		render(<Navbar />);
	});
	it('should contain logo image', () => {
		const logoImage = screen.getByRole('img');
		expect(logoImage).toBeInTheDocument();
	});
	it('should contain home link', () => {
		const homeLinkText = screen.getByText(/home/i);
		expect(homeLinkText).toBeInTheDocument();
	});
	it('should contain blog link', () => {
		const blogLinkText = screen.getByText(/blog/i);
		expect(blogLinkText).toBeInTheDocument();
	});
	it('should contain link', () => {
		const aboutLinkText = screen.getByText(/about/i);
		expect(aboutLinkText).toBeInTheDocument();
	});
	it('should contain about button', () => {
		const contactButton = screen.getByText(/contact/i);
		expect(contactButton).toBeInTheDocument();
	});
});
