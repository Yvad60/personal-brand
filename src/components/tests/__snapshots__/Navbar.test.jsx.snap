// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar test should render the Navbar without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-full bg-primary_light flex text-primary_dark font-bold text-xl justify-between py-5 shadow-main px-20"
      >
        <img
          alt="logo"
          src="test-file-stub"
        />
        <ul
          class="w-1/2 flex justify-between"
        >
          <li>
            Home
          </li>
          <li>
            Blog
          </li>
          <li>
            About
          </li>
          <button
            type="button"
          >
            Contact
          </button>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-full bg-primary_light flex text-primary_dark font-bold text-xl justify-between py-5 shadow-main px-20"
    >
      <img
        alt="logo"
        src="test-file-stub"
      />
      <ul
        class="w-1/2 flex justify-between"
      >
        <li>
          Home
        </li>
        <li>
          Blog
        </li>
        <li>
          About
        </li>
        <button
          type="button"
        >
          Contact
        </button>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
